tests/DataHandling/XMLExtractorTest.php

```diff
@@ -0,0 +1,259 @@
+    
+    /**
+     * @var string XML source with element
+     */
+    const ATTRIBUTE_IN_CHILD = '<?xml version="1.0" encoding="UTF-8"?>
+        <elements>
+          <element>
+            <element1 attribute1="at1">el1</element1>
+            <element2 attribute1="at2">el2</element2>
+            <element3>el3</element3>
+            <element4>el4</element4>
+          </element>
+          <element>
+            <element1 attribute1="at3">el11</element1>
+            <element2 attribute1="at4">el22</element2>
+            <element3>el33</element3>
+            <element4>el44</element4>
+          </element>  
+        </elements>';
+    
+    protected $attributeInChildTransformations = array('element1/@attribute1' => '',
+        'element2/@attribute1' => '',
+        );
+    
+    protected $attributeInChildExpectedReadArray = array(array(
+        'element1/@attribute1' => 'at1',
+        'element2/@attribute1' => 'at2',
+    ),
+    array(
+        'element1/@attribute1' => 'at3',
+        'element2/@attribute1' => 'at4',
+    ));
+    
+    /* end of data for testProcessXMLWithAttributeInChild */
+    /* data for testProcessXMLSimpleTranslation */
+    
+    protected $simpleTranslationExpectedReadArray = array(array(
+        'element_first' => 'el1',
+        'element2' => 'el2',
+        'element3' => 'el3',
+        'element4' => 'el4',
+    ),
+    array(
+        'element_first' => 'el11',
+        'element2' => 'el22',
+        'element3' => 'el33',
+        'element4' => 'el44',
+    ));
+    
+    /* end of data for testProcessXMLSimpleTranslation */
+    /* data for testProcessXMLSimpleReplaceSourceElement */
+    
+    protected $replaceSourceElementExpectedReadArray = array(array(
+        'dynamicElementName' => 'el1',
+        'element2' => 'el2',
+        'element3' => 'el3',
+        'element4' => 'el4',
+    ),
+    array(
+        'dynamicElementName' => 'el11',
+        'element2' => 'el22',
+        'element3' => 'el33',
+        'element4' => 'el44',
+    ));
+    
+    /* end of data for testProcessXMLSimpleReplaceSourceElement */
+    /* data for testProcessXMLEmptyElement */
+    
+    const EMPTY_ELEMENT_SOURCE = '<?xml version="1.0" encoding="UTF-8"?>
+        <elements>
+          <element>
+            <element1/>
+            <element2>el2</element2>
+            <element3></element3>
+            <element4>el4</element4>
+          </element>
+          <element>
+            <element1></element1>
+            <element2/>
+            <element3>el33</element3>
+            <element4>el44</element4>
+          </element>  
+        </elements>';
+    
+    protected $emptyElementExpectedReadArray = array(array(
+        'element1' => '',
+        'element2' => 'el2',
+        'element3' => '',
+        'element4' => 'el4',
+    ),
+    array(
+        'element1' => '',
+        'element2' => '',
+        'element3' => 'el33',
+        'element4' => 'el44',
+    ));
```

Comment:
But what happens then? (In case this happens by accident.) Does it raise an error? Return some data (what kind of)? Return none?
Is this maybe worth a test?